core/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/status/TransportSnapshotsStatusAction.java

```diff
@@ -197,20 +198,20 @@ private SnapshotsStatusResponse buildResponse(SnapshotsStatusRequest request, Li
         if (Strings.hasText(request.repository())) {
             if (request.snapshots() != null && request.snapshots().length > 0) {
                 for (String snapshotName : request.snapshots()) {
-                    SnapshotId snapshotId = new SnapshotId(request.repository(), snapshotName);
-                    if (currentSnapshotIds.contains(snapshotId)) {
+                    Snapshot snapshot = new Snapshot(request.repository(), snapshotName);
+                    if (currentSnapshotIds.contains(snapshot)) {
                         // This is a snapshot the is currently running - skipping
                         continue;
                     }
-                    SnapshotInfo snapshot = snapshotsService.snapshot(snapshotId);
+                    SnapshotInfo snapshotInfo = snapshotsService.snapshot(snapshot);
```

Comment:
As we don't know the snapshot uuid here, we re-resolve the list of snapshots for every entry in `request.snapshots()`. Can we do that only once?